File: Csharp/Lab8/Lab8/Lab8/cng/Model/Entity/Container/JSFContainer/JSFArrayTree.cs
using System.Collections;
using System.Collections.Generic;
using Lab8.cng.Model.Entity.Abstracts;

namespace Lab8.cng.Model.Entity.Container.JSFContainer
{
    public class JSFArrayTree : IJsfTree
    {
        
        private ArrayList _toys = new ArrayList();
        
        public IEnumerator<Toy> GetEnumerator()
        {
            return (IEnumerator<Toy>) _toys.GetEnumerator();
        }

        IEnumerator IEnumerable.GetEnumerator()
        {
            return _toys.GetEnumerator();
        }

        public int GetSizeOfTree()
        {
            return _toys.Count;
        }

        public void Remove(int index)
        {
            _toys.RemoveAt(index);
        }

        public void ChangeToy(Toy toy, int index)
        {
            _toys.RemoveAt(index);
            _toys.Insert(index,toy);
        }

        public void Add(Toy toy)
        {
            _toys.Add(toy);
        }

        public Toy[] GetMasOfObjects()
        {
            return (Toy[]) _toys.ToArray();
        }

        public Toy GetToy(int index)
        {
            return (Toy) _toys.GetRange(index, 1).ToArray()[0];
        }
    }
}

File: Java/Desktop/NewLab8/src/by/bntu/fitr/povt/cng/model/entity/container/JSFContainer/JSFArrayTree.java
package by.bntu.fitr.povt.cng.model.entity.container.JSFContainer;

import by.bntu.fitr.povt.cng.model.entity.abstracts.Toy;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Iterator;

public class JSFArrayTree implements JSFTrees{

    private ArrayList<Toy> toys = new ArrayList<>();

    public JSFArrayTree(Toy...ts) {
        toys.addAll(Arrays.asList(ts));
    }

    public JSFArrayTree(){

    }

    public Toy[] getMasOfObjects(){
        return (Toy[]) toys.toArray();
    }

    public Toy getToy(int index){
        return toys.get(index);
    }

    public void changeToy(Toy toy, int index){
        toys.remove(index);
        toys.add(index, toy);
    }

    public void remove(int index){
        toys.remove(index);
    }

    public void add(Toy toy){
        toys.add(toy);
    }

    public int getSizeOfTree(){
        return toys.size();
    }

    @Override
    public int hashCode() {
        int result = 0;
        for (Toy toy : toys){
            result += toy.hashCode();
        }
        return result;
    }

    @Override
    public boolean equals(Object obj) {
        if (obj == null && !(obj instanceof JSFArrayTree)){
            return false;
        }
        return obj == this && this.hashCode() == obj.hashCode() && obj.getClass() == this.getClass();
    }

    @Override
    public String toString() {
        return toys.toString();
    }

    @Override
    public Iterator<Toy> iterator() {
        return toys.iterator();
    }
}
